src/coreclr/src/jit/codegenarmarch.cpp

```diff
@@ -1402,7 +1402,7 @@ void CodeGen::genMultiRegStoreToLocal(GenTree* treeNode)
         // Insert pieces in reverse order
         for (int i = regCount - 1; i >= 0; --i)
         {
-            var_types type = op1->GetRegTypeByIndex(i);
+            var_types type = op1->gtSkipReloadOrCopy()->GetRegTypeByIndex(i);
```

Comment:
A `GenTreeCopyOrReload` doesn't have reg types, so we need to skip it to get the types from its source.